tests/test_transform.py

```diff
@@ -506,3 +506,13 @@ def test_2421_rpc_height_ignored():
         x2, y2 = src.xy(0, 0, z=2000, transform_method=transform_method)
         assert abs(x2 - x1) > 0
         assert abs(y2 - y1) > 0
+
+def test_tps_transformer():
```

Comment:
@cpegel its not trivial to test!

If you agree with the suggestion below, will you add some notes that will help us remember why we're testing in this manner?

```suggestion
def test_gcp_transformer_tps_option():
    """Use thin plate spline transformation when requested."""
    # TPS ensures that GCPs are (to within some precision) solutions of the transformation.
    # This is not the case for polynomials transformations. 
```

Reply:
> If you agree with the suggestion below, will you add some notes that will help us remember why we're testing in this manner?

I think your comment describes it pretty well: While the polynomial approach tries to fit one polynomial transform to the whole image, the TPS approach is a piecewise defined (spline) transformation that ensures the GCPs stay at the specified geolocations. Testing the accuracy of TPS between GCPs is of course more complicated and that should be done in GDAL itself, since we are just wrapping GDAL here. Hence, this test just checks if creating, using and destroying the transformer works and loops over the GCPs to check if they do indeed stay at the specified geolocations.